game-service/src/main/java/org/libertybikes/game/core/GameRound.java

```diff
@@ -483,20 +483,11 @@ public void startGame() {
         // Issue a countdown to all of the clients
         gameState = State.STARTING;
 
-        sendToClients(clients.keySet(), new OutboundMessage.StartingCountdown(STARTING_COUNTDOWN));
-        delay(TimeUnit.SECONDS.toMillis(STARTING_COUNTDOWN));
-
-        paused.set(false);
-        for (Player p : getPlayers())
-            if (STATUS.Connected == p.getStatus())
-                p.setStatus(STATUS.Alive);
-        broadcastPlayerList();
-        if (!gameRunning.get()) {
-            ManagedScheduledExecutorService exec = executor();
-            if (exec != null)
-                exec.submit(this);
+        ManagedScheduledExecutorService exec = executor();
+        if (exec != null) {
+            Starter start = new Starter(this);
+            exec.submit(start);
```

Comment:
is it necessary to offload this to a separate thread? It should be fine that we can't process any other messages on this thread during the countdown, since we don't allow players to switch directions until the game starts anyway?

Reply:
If i didn't offload into another thread, it would back up and process them all after start returned, making it look like i changed nothing.